plans/gmail-oauth-authentication.md

```diff
@@ -0,0 +1,1979 @@
+        String registrationId = userRequest.getClientRegistration().getRegistrationId();
+        OAuth2UserInfo userInfo = OAuth2UserInfoFactory.getOAuth2UserInfo(registrationId, oauth2User.getAttributes());
+        
+        // Check if account exists using local query model
+        Optional<AccountQueryModelState> existingAccount = 
+            accountQueryService.findByOAuthProvider(userInfo.getProvider(), userInfo.getId());
+        
+        if (existingAccount.isEmpty()) {
+            // New user - create account via Commands service using AkcesClient
+            String userId = UUID.randomUUID().toString();
+            CreateAccountCommandV2 command = new CreateAccountCommandV2(
+                userId,
+                "Unknown",  // Country - can be enhanced later
+                userInfo.getName(),
+                "",  // Last name - extract if available
+                userInfo.getEmail(),
+                userInfo.getProvider(),  // e.g., "google"
+                userInfo.getId()         // Provider-specific user ID
+            );
+            
+            // Send command to Commands service
+            akcesClient.send("DEFAULT_TENANT", command)
+                .toCompletableFuture()
+                .join();  // Wait for account creation
+            
+            // Return OAuth2User with userId for JWT generation
+            return new DefaultOAuth2User(
+                oauth2User.getAuthorities(),
+                Map.of(
+                    "sub", userId,
+                    "email", userInfo.getEmail(),
+                    "name", userInfo.getName(),
+                    "oauth_provider", userInfo.getProvider(),
+                    "oauth_provider_id", userInfo.getId()
+                ),
+                "sub"
+            );
+        } else {
+            // Existing user - return with existing userId
+            AccountQueryModelState account = existingAccount.get();
+            return new DefaultOAuth2User(
+                oauth2User.getAuthorities(),
+                Map.of(
+                    "sub", account.getUserId(),
+                    "email", account.getEmail(),
+                    "name", account.getFirstName(),
+                    "oauth_provider", account.getOauthProvider(),
+                    "oauth_provider_id", account.getOauthProviderId()
+                ),
+                "sub"
+            );
+        }
+    }
+}
+```
+
+**AccountQueryService (in Auth service):**
+
+```java
+@Service
+public class AccountQueryService {
+    
+    private final AccountQueryModel accountQueryModel;
+    
+    @Autowired
+    public AccountQueryService(AccountQueryModel accountQueryModel) {
+        this.accountQueryModel = accountQueryModel;
+    }
+    
+    public Optional<AccountQueryModelState> findByOAuthProvider(String provider, String providerId) {
+        // Query local cache maintained by AccountQueryModel
+        return accountQueryModel.findByOAuthProvider(provider, providerId);
```

Comment:
this is not possible using the kafka based QueryModel, this implemenation will have to be based on the JdbcDatabaseModel and then queries like this can be implemented. Please update the plan accordingly